File: 2644. Find the Maximum Divisibility Score/Solution.py
class Solution:
    def maxDivScore(self, nums: List[int], divisors: List[int]) -> int:
        high, res = 0, 1e9 + 1
        for d in divisors:
            curr = 0
            for n in nums:
                if n % d == 0: curr += 1
            if curr > high:
                res = d
                high = curr
            elif curr == high:
                res = min(d, res)
        return res

File: 2644. Find the Maximum Divisibility Score/Solution.java
class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        int high = 0, res = 1000000001;
        for (int d : divisors) {
            int curr = 0;
            for (int n : nums) {
                if (n % d == 0) curr++;
            }
            if (curr > high) {
                res = d;
                high = curr;
            } else if (curr == high) {
                res = Math.min(res, d);
            }
        }
        return res;
    }
}